internal/driver/onvifdiscovery_test.go

```diff
@@ -0,0 +1,349 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package driver
+
+import (
+	"testing"
+
+	sdkModel "github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+	contract "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func createTestDeviceList() []contract.Device {
+	return []models.Device{
+		{
+			Name: "device-onvif-camera", Protocols: map[string]models.ProtocolProperties{
+				OnvifProtocol: map[string]string{
+					EndpointRefAddress: "abc",
+				},
+			},
+		},
+		{
+			Name: "testDevice1", Protocols: map[string]models.ProtocolProperties{
+				OnvifProtocol: map[string]string{
+					EndpointRefAddress: "123",
+				},
+			},
+		},
+		{
+			Name: "testDevice2", Protocols: map[string]models.ProtocolProperties{
+				OnvifProtocol: map[string]string{
+					EndpointRefAddress: "456",
+				},
+			},
+		},
+		{
+			Name: "testDevice3", Protocols: map[string]models.ProtocolProperties{
+				OnvifProtocol: map[string]string{
+					EndpointRefAddress: "789",
+				},
+			},
+		},
+	}
+}
+
+func createDiscoveredList() []sdkModel.DiscoveredDevice {
+	return []sdkModel.DiscoveredDevice{
+		{
+			Name: "testDevice1", Protocols: map[string]models.ProtocolProperties{
+				OnvifProtocol: map[string]string{
+					EndpointRefAddress: "123",
+				},
+			},
+		},
+		{
+			Name: "testDevice2", Protocols: map[string]models.ProtocolProperties{
+				OnvifProtocol: map[string]string{
+					EndpointRefAddress: "456",
+				},
+			},
+		},
+		{
+			Name: "testDevice3", Protocols: map[string]models.ProtocolProperties{
+				OnvifProtocol: map[string]string{
+					EndpointRefAddress: "789",
+				},
+			},
+		},
+	}
+}
+
```

Comment:
Comment to briefly describe what this unit test achieves to be consistent with other unit tests.